src/server/services/controllers/accountController.js

```diff
@@ -13,47 +14,85 @@ exports.getAllAccounts = function(req, res, next) {
     models.getAllAccounts(function(err, accounts) {
         if (err) {
             sendError(400, res, err, 'Failed on getAllAccounts');
+            return next;
         }
-        res.json(accounts);
+        res.status(200).json(accounts);
     });
 }
 
 exports.getAccountById = function(req, res, next) {
+    if (!req.params && !req.params.id) {
+        sendError(404, res, "", "Account not found");
+        return next;
+    }
     console.log('getAccountById ' + req.params.id);
-    models.getAccountById(req.params.id, function(err, accounts) {
+    models.getAccountById(req.params.id, function(err, account) {
         if (err) {
             sendError(404, res, err, 'Failed on getAccountById');
+            return next;
         }
-        res.json(accounts);
+        if (account === null) {
```

Comment:
This should probably be
```javascript
if (!account) {
```
to cover the cases in which `account` is `undefined` (which is not `null`). 
```javascript
var account;
undefined
account
undefined
(account === null)
false
(account == null)
true
(!account)
true
```